plugins/ltac/extraargs.mlg

```diff
@@ -147,8 +147,10 @@ let pr_occurrences = pr_occurrences () () ()
 
 let pr_gen prc _prlc _prtac c = prc c
 
+let gcc () = let env = Global.env () in Evd.from_env env, env
+
 let pr_globc _prc _prlc _prtac (_,glob) =
-  let _, env = Pfedit.get_current_context () in
+  let _, env = gcc () in
```

Comment:
Why this change?

Reply:
We could have the printers use the goal info if any, but I am not sure how we can integrate this into `ARGUMENT EXTEND`.

Reply:
What is the context such a printer is called from?

Reply:
I concur with @herbelin , the code picks the wrong env. It was a very bad choice to reuse the same type, IMO, but the system is coded this way. Let me explain. `env` can be the global one, or the proof one (including local hyps). The `named_context` part of the env is used to both represent section variables (that are somewhat global) and local hyps during a proof. The old code was taking the `env` of the first goal, that possibly includes some hyps. You take `Global.env` that has no hyps.

In the new monadic API you will see that there are two ways to get an env, one from the goal and one from nothing (the global one, stored in the monad state with the hope that it would help implementing backtracking safe)

Reply:
Indeed, this can be fixed now with the new coqpp mode.